### ISSUE TITLE
1.1.0: crypto\ct\ct_log.c(143): error C2065: "ct_log_list": undeclared Identifier

### ISSUE THREAD

**Issue by MeerMusik** [openssl/openssl#4458]
Hello.

With the current 1.1.0 Stable Branch I get the following Error:

```
cl  /I "." /I "crypto\include" /I "include" -DOPENSSL_USE_APPLINK -DDSO_WIN32 -DNDEBUG -DOPENSSL_THREADS -DOPENSSL_NO_STATIC_ENGINE -DOPENSSL_PIC -DOPENSSL_IA32_SSE2 -DOPENSSL_BN_ASM_MONT -DOPENSSL_BN_ASM_MONT5 -DOPENSSL_BN_ASM_GF2m -DSHA1_ASM -DSHA256_ASM -DSHA512_ASM -DRC4_ASM -DMD5_ASM -DAES_ASM -DVPAES_ASM -DBSAES_ASM -DGHASH_ASM -DECP_NISTZ256_ASM -DPADLOCK_ASM -DPOLY1305_ASM "-DENGINESDIR=\"C:\\openssl11x64\\lib\\engines-1_1\"" "-DOPENSSLDIR=\"C:\\openssl11x64\\\"" -W3 -wd4090 -Gs0 -GF -Gy -nologo -DOPENSSL_SYS_WIN32 -DWIN32_LEAN_AND_MEAN -DL_ENDIAN -D_CRT_SECURE_NO_DEPRECATE -DUNICODE -D_UNICODE /MD /O2 /Zi /Fdossl_static -c /Focrypto\ct\ct_log.obj "crypto\ct\ct_log.c"
ct_log.c
crypto\ct\ct_log.c(143): error C2065: "ct_log_list": nichtdeklarierter Bezeichner
crypto\ct\ct_log.c(143): error C2224: Der linke Teil von ".cnf" muss eine Struktur/Union sein
crypto\ct\ct_log.c(143): error C2001: Zeilenvorschub in Konstante.
crypto\ct\ct_log.c(143): error C2143: Syntaxfehler: Es fehlt ";" vor "Zeichenfolge"
NMAKE : fatal error U1077: ""C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\VC\Tools\MSVC\14.11.25503\bin\HostX64\x64\cl.EXE"": Rückgabe-Code "0x2"
Stop.
NMAKE : fatal error U1077: ""C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\VC\Tools\MSVC\14.11.25503\bin\HostX64\x64\nmake.exe"": Rückgabe-Code "0x2"
Stop.
```

Software used:
* Windows 10 15063
* VS 2017 15.3 Community Edition
* NASM 2.13.01, 2.13.02 RC2
* OpenSSL latest 1.1.0 Stable Branch
* Perl: 5.24.x, 5.26.x

Tested OpenSSL Configure Options:
* perl Configure no-comp no-rc5 no-idea no-mdc2 no-tls1 no-tls1-method no-tls1_1 no-tls1_1-method no-dtls no-dtls1-method no-dtls1_2 no-dtls1_2-method no-deprecated no-weak-ssl-ciphers VC-WIN64A --prefix=%SystemDrive%\openssl11x64\ --openssldir=%SystemDrive%\openssl11x64\
* perl Configure VC-WIN64A --prefix=%SystemDrive%\openssl11x64\ --openssldir=%SystemDrive%\openssl11x64\

Only a OpenSSL User but looking through the latest Commits, is it possible that
* https://github.com/openssl/openssl/commit/de6db95d7bc9a0f8993d4e125028a02a7ec46e96
is causing this? Well on second thought: It is just a Null Pointer Fix. But yeah OpenSSL internals are still so over my head :)

Thanks!
Oliver

EDIT01: Forgot to mention: Last successful compilation were on September 22nd and 23rd. No builds after that until I tried again on October 2nd, where I first ran into that Issue.

**Comment by levitte** [comment]
Nope, that commit isn't it. But if you look at line 143, where the error was found, you can see this:

```
      fpath = CTLOG_FILE;
```

Looks innocent, right? However, CTLOG_FILE is a macro, and if it isn't defined, that line is effectively this:

```
      fpath = ;
```

that would give the error you experience. 

I'll investigate...

**Comment by levitte** [comment]
Alternatively, from a second look at the error message (my German is *really* bad), it might be that CTLOG_FILE  is badly defined

**Comment by MeerMusik** [comment]
Interesting. Thanks for the Explanation - and of course taking a look :)

**Comment by levitte** [comment]
Ok, I think I know what's going on.  How about skipping that ending backslash in the directory specs in your configuration?  I.e. do this:

```
perl Configure no-comp no-rc5 no-idea no-mdc2 no-tls1 no-tls1-method no-tls1_1 no-tls1_1-method no-dtls no-dtls1-method no-dtls1_2 no-dtls1_2-method no-deprecated no-weak-ssl-ciphers VC-WIN64A --prefix=%SystemDrive%\openssl11x64 --openssldir=%SystemDrive%\openssl11x64
```

```
perl Configure VC-WIN64A --prefix=%SystemDrive%\openssl11x64 --openssldir=%SystemDrive%\openssl11x64
```

Those are the configurations you showed above with just a tiny edit.

**Comment by MeerMusik** [comment]
Tested only the 1st one: It compiles like expected again and all Tests pass (so the second one should work too - can still test it if necessary). I am having this backslash for so long in that Parameter, I did not even think about testing it -.-

Does this mean this BR is invalid, as it is an User Error and that "backslash thing" was fixed in the OpenSSL Branch through whatever Commit?

**Comment by levitte** [comment]
I don't believe we should "blame the user" for this.  Having an ending backslash is perfectly legitimate, it's just our config mechanisms that don't handle it right.

**Comment by levitte** [comment]
All you did just now was to help verify my theory.  Thank you.

**Comment by levitte** [comment]
I think I've corrected the problem.  Have a look at #4461

**Comment by MeerMusik** [comment]
‎Thanks. Will take a look later this (late) evening (GMT+2) - no time right now.                                                                                                                                                                                                                                                                                                                                        Sent from a Mobile Device. Encoding Problems can happen.                                                                                                                                                                                                                Von: Richard LevitteGesendet: Mittwoch, 4. Oktober 2017 10:41An: openssl/opensslAntwort an: openssl/opensslCc: Oliver Niebuhr; AuthorBetreff: Re: [openssl/openssl] 1.1.0: crypto\ct\ct_log.c(143): error C2065: "ct_log_list": undeclared Identifier (#4458)I think I've corrected the problem.  Have a look at #4461

—You are receiving this because you authored the thread.Reply to this email directly, view it on GitHub, or mute the thread.


  
  




{"api_version":"1.0","publisher":{"api_key":"05dde50f1d1a384dd78767c55493e4bb","name":"GitHub"},"entity":{"external_key":"github/openssl/openssl","title":"openssl/openssl","subtitle":"GitHub repository","main_image_url":"https://cloud.githubusercontent.com/assets/143418/17495839/a5054eac-5d88-11e6-95fc-7290892c7bb5.png","avatar_image_url":"https://cloud.githubusercontent.com/assets/143418/15842166/7c72db34-2c0b-11e6-9aed-b52498112777.png","action":{"name":"Open in GitHub","url":"https://github.com/openssl/openssl"}},"updates":{"snippets":[{"icon":"PERSON","message":"@levitte in #4458: I think I've corrected the problem.  Have a look at #4461"}],"action":{"name":"View Issue","url":"https://github.com/openssl/openssl/issues/4458#issuecomment-334088345"}}}
